src/services/digital_out.py

```diff
@@ -1,11 +1,32 @@
 from src.shared_memory import SharedMemory
 from src.pin.pinout import Pinout
 import src.pin.memory as memory
+from src.test_lib.condition import Condition
+import asyncio
 class DigitalOutService:
     def __init__(self, pin: Pinout):
         self._pin = SharedMemory.get_pin(pin, memory.PinType.DigitalOut)
         
     def get_pin_state(self) -> memory.DigitalOut.State:
         return self._pin.data.state
 
+    class WaitForStateCondition(Condition):
+        def __init__(self, digitalOut, cond):
+            self._digitalOut = digitalOut
+            self._cond = cond
+        
+        async def check(self) -> bool:
+            while not self._cond(self._digitalOut.get_pin_state()):
+                await asyncio.sleep(0)
+                pass
+            return True
+
+    def wait_for_state(self, cond: function) -> Condition:
+        return self.WaitForStateCondition(self, cond)
+    
+    def wait_for_high(self) -> Condition:
+        return self.WaitForStateCondition(self,lambda x: x == memory.DigitalOut.State.High)
     
+    def wait_for_low(self) -> Condition:
+        return self.WaitForStateCondition(self,lambda x: x == memory.DigitalOut.State.Low)
+    
```

Comment:
see, now state refers to the digital out actual state, this is a source of confusion, leave these names as they are, because they are actually the state of the digital out, but change all the other functions with the same name but checking instead `is_on`